src/plugins/data/public/ui/dataset_selector/index.tsx

```diff
@@ -18,9 +18,21 @@ const ConnectedDatasetSelector = ({ onSubmit }: ConnectedDatasetSelectorProps) =
   const { services } = useOpenSearchDashboards<IDataPluginServices>();
   const queryString = services.data.query.queryString;
   const [selectedDataset, setSelectedDataset] = useState<Dataset | undefined>(
-    () => queryString.getQuery().dataset || queryString.getDefaultQuery().dataset
+    () => queryString.getQuery().dataset
   );
 
+  useEffect(() => {
+    const subscription = queryString.getUpdates$(0).subscribe((query: Query) => {
+      if (query.dataset !== selectedDataset) {
```

Comment:
We recently removed this `useEffect()` in #8347 to avoid duplicate calls. Does this `if` condition avoid the duplicate calls? Are we adding this back in so it picks up updates from state sync?

Reply:
@sejli The reasoning behind adding this , is when we are using to load the state from the saved Object it loads it without using the dataset selector component. Therefore although the actual query state has been updated but since the dataset selector is not triggered, the dataset selector content is not updated. The if condition is to avoid multiple updates.